README.md

```diff
@@ -1,2 +1,186 @@
+
+To avoid Redux code base from growing, inconsistency and lowering maintainability, we created this lightweight library (<4Kb) that is a factory of reducers, actions and selectors.
+
+## Installation
+ - `yarn add trampss-redux-data-store`
+ - `npm i trampss-redux-data-store`
+
+### peer dependency
+ - `lodash` : we use the minimum of lodash function trying to have a lightweight webpack bundle.
+   - `keyBy`
+   - `without`
+   - `uniq`
+   - `omit`
+   - `at`
+
+## API
+ - [factory](#factory)
+ - [actions](#actions)
+ - [selectors](#selectors)
+
+### factory
+You need to use the factory to get a new set of reducer/actions/selectors :
+```es6
+import factory from 'trampss-redux-data-store'
+```
+
+This factory takes three parameters `factory(fieldKey)(path)(name)`:
+ - **fieldKey** (mandatory), the field used to identify your objects (`id` for example)
+   - you have to set this parameter.
+ - **path** (optional), where the reducer will be combined via `combineReducer`
+   - if empty `export default factory('id')()('todos')`, the reducer will be register ate the root level of the redux state
+   - you can use dot notation, like `api.raw`: your reducer will be combined into `state.api.raw.<your_reducer>`
+ - **name** (mandatory), the reducer name (for instance: `todos`)
+   - it's used to generate actions types
+   - it's used to retrieve informations from selectors
+
+Example:
+ - this reducer will use `id` as key field
+ - it's combined into `state.api.raw`
+ - its name is `todos`
+```es6
+import factory from 'trampss-redux-data-store'
+// factory(fieldKey)(path)(name)
+export default factory('id')('api.raw')('todos')
+```
+
+Data will be stored into `state.api.raw.todos`
+
+### reducer
+The previous factory returns a function which is a reducer.
+You just have to combine it like any other reducer :
+```es6
+import { createStore, combineReducers, compose, applyMiddleware } from 'redux'
+
+// import your reducer
+// (created by tramps-redux-data-store factory)
+import todos from './myTodosReducer'
+
+// create your Redux store as usual
+const store = createStore(
+  combineReducers({
+    // [other reducer]
+    api: combineReducers({
+      // [other reducer]
+      raw: combineReducers({
+        // import your reducer into api.raw
+        // since we configured this path
+        todos,
+      }),
+      // [other reducer]
+    }),
+    // [other reducer]
+  }),
+  /* your Redux middlewares */
+)
+
+export default store
+```
+
+### actions
+The factory returns a function (this is the reducer) that also contains actions and selectors as fields.
+Some generic actions are available. By now, it's not possible to add custom ones.
+
+Actions are:
+ - `set` : set an array of instances of your resource
+   - `set(<array>)`
+   - will generate this kind of Redux action:
+     - `{ type: 'SET_todos', payload: <array> }`
```

Comment:
You may need to add also the path, to avoid conflicts like that : 
`factory('id')('search')('users')`  =>  action `SET_users`
`factory('id)('followers')('users')`  =>  action `SET_users`